[4.7.7] starting measurements automatically - Cleanups (patch 3/3)

vitalii-komisarenko opened the pull request:
Because of its size PR #1139 needs to be split.
Description: #1139 (review)

This PR reflects the cleanup phase of the change.

This PR depends on #1290 and #1289 in order to work properly.

vitalii-komisarenko committed:
tlvf: Remove unused 'channel' and 'vap_id' field

The 'channel' and 'vap_id' fields are no longer used or needed so they are removed
from START_MONITORING message.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

vitalii-komisarenko committed:
agent: remove CLIENT_STOP_MONITORING message

Remove all CLIENT_STOP_MONITORING_REQUEST message sending and handlers
from the code because it is replaced with an internal STA_Disconnected BWL
event handler on the monitor thread.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

tomereli commented:
Please name the PRs so we will know which one should be reviewed first:
For example:
[4.7.7] starting measurements automatically - BWL (patch 1/3)
[4.7.7] starting measurements automatically - Monitor/Slave (patch 2/3)
[4.7.7] starting measurements automatically - Cleanups (patch 3/3)

tomereli commented on agent/src/beerocks/monitor/monitor_thread.cpp:
```diff
@@ -772,64 +772,6 @@ bool monitor_thread::handle_cmdu_vs_message(Socket *sd, ieee1905_1::CmduMessageR
         received_error_notification_ack_retry = 0;
         break;
     }
-    case beerocks_message::ACTION_MONITOR_CLIENT_START_MONITORING_REQUEST: {
```
This has to be tested on RDKB/UGW - since UTAF is not yet ready I suggest @kantera800 you test it.

kantera800 replied:
@tomereli @vitalii-komisarenko 
This PR is currently fails to build on RDKB/UGW platform so i cant test it.

_/home/akanterx/work/dev2/rdkb/atom_rdkbos/build/tmp/work/core2-32-rdk-linux/prplmesh/git+f8786935bdc0b58021710248bf2620896b74c44f-r0/git/agent/src/beerocks/monitor/monitor_thread.cpp:789:66: error: 'beerocks_message::sClientMonitoringParams {aka struct beerocks_message::sClientMonitoringParams}' has no member named 'vap_id'
          int vap_id                       = int(request->params().vap_id);_

morantr replied:
I was able to build it successfully on RDKB and UGW.

tomereli reviewed (commented):
This PR is a bit more risky, it removes the VS path so has to be tested on RDKB and UGW by Intel team to make sure it doesn't break anything, so not approving **yet**.
@morantr @kantera800 please provide a version for @amritak23 or @Mila-Sh and ask for a QA test.
We can wait a bit with this PR but we do want to merge it in a couple of days max.

vitalii-komisarenko committed:
tlvf: Remove CLIENT_STOP_MONITORING_REQUEST messages from the code

Including auto-generated code.

Signed-off-by: Vitalii Komisarenko <vitalii.komisarenko@gmail.com>

morantr committed:
agent: Fix crash

The line changed on this commit has been "if (client != nullptr)" in the past.
On PR #1289 it was moved from start monitoring message handler to sta_connected handler but mistakenly inverted the condition.
Access pointer which is nullptr causing to a crash.

Invert the condition back.

Signed-off-by: Moran Shoeg <moranx.shoeg@intel.com>

kantera800 reviewed (approved):
This PR  fixes start monitoring .
 Also,  @moran tested it on UGW/RDKB so approving.
